### PR TITLE
[UI] Configurations UI Enhancements | Apps | Designs | Filters

### PR DESCRIPTION
**Pull request by Chadha93**
Signed-off-by: Gaurav Chadha <gauravchadha1676@gmail.com>
**Description**
Configurations UI Enhancements | Apps | Designs | Filters

This PR fixes #

- [x] Adds source type icon in grid view cards dynamically.
- [x] Adds an icon with a tooltip in the source type column to indicate to the user what action can be performed on icon click—[UX].
- [x] Fixes responsive issue with 3 buttons on designs grid view cards.
- [x] Makes Undeploy and Deploy icon scale to the same ratio.
- [x] Applied standard and hovered state for undeploy red button according to the decided color
- [x] Fixes all empty state bugs with import.
- [x] Changed import modal and prompt components title background color to desired.
- [x] Made grid styles resuable.   

**Notes for Reviewers**

https://user-images.githubusercontent.com/65453826/184954013-08a47443-ea08-4c4a-90fd-71b28475d441.mp4

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
This is looking good. Worth noting that this also brings the grid-based view on par with the table view in terms of functionality.

**Review by leecalcote — DISMISSED** [review]

File: ui/components/MesheryApplications.js [hunk 1]
```diff
@@ -606,7 +607,12 @@ function MesheryApplications({
         customHeadRender : function CustomHead({ index, ...column }) {
           return (
             <TableCell key={index}>
-              <b>{column.label}</b>
+              <Tooltip title="click source type to download application">
```
Comment (asubedy) [inline ui/components/MesheryApplications.js:610]:
```suggestion
              <Tooltip title="Click source type to download Application">
```

File: ui/components/MesheryApplications/ApplicationsCard.js [hunk 2]
```diff
@@ -70,9 +73,19 @@ function MesheryApplicationCard({
...
           <div>
-            <Typography variant="h6" component="div">
-              {name}
-            </Typography>
+            <div style={{ display : "flex", justifyContent : "space-between" }}>
+              <Typography variant="h6" component="div">
+                {name}
+              </Typography>
+              <IconButton
+                title="click to download"
```
Comment (asubedy) [inline ui/components/MesheryApplications/ApplicationsCard.js:81]:
```suggestion
                title="Click to Download"
```

File: ui/components/MesheryPatterns/MesheryPatternCard.js [hunk 3]
```diff
@@ -93,6 +93,7 @@ function MesheryPatternCard({
             <div className={classes.cardButtons} >
 
               <Button
+                title="undeploy"
```
Comment (asubedy) [inline ui/components/MesheryPatterns/MesheryPatternCard.js:96]:
```suggestion
                title="Undeploy"
```

File: ui/components/MesheryPatterns/MesheryPatternCard.js [hunk 4]
```diff
@@ -104,17 +105,19 @@ function MesheryPatternCard({
               </Button>
 
               <Button
+                title="deploy"
```
Comment (asubedy) [inline ui/components/MesheryPatterns/MesheryPatternCard.js:108]:
```suggestion
                title="Deploy"
```

File: ui/components/MesheryPatterns/MesheryPatternCard.js [hunk 5]
```diff
@@ -104,17 +105,19 @@ function MesheryPatternCard({
...
                 }
                 className={classes.testsButton}
               >
                 <DoneAllIcon className={classes.iconPatt} />
-                Deploy
+                <span className={classes.btnText}>Deploy</span>
               </Button>
 
               <Button
+                title="design"
```
Comment (asubedy) [inline ui/components/MesheryPatterns/MesheryPatternCard.js:120]:
```suggestion
                title="Design"
```

**Review by asubedy — CHANGES_REQUESTED** [review]
Some typography changes. Rest looks amazing 👍

File: ui/components/MesheryApplications/ApplicationsCard.js (hunk 2 above)
Comment (Chadha93) [reply ui/components/MesheryApplications/ApplicationsCard.js:81]: let's keep this one in lowercase only.

Comment (asubedy) [reply ui/components/MesheryApplications/ApplicationsCard.js:81]: "c" of click?

**Comment by legendarykamal** [comment]
@Chadha93 great work :clap:  
just wanted to ask is that multiple files downloading intentional at 0:14 of the video ?

**Comment by Chadha93** [comment]
> @Chadha93 great work clap just wanted to ask is that multiple files downloading intentional at 0:14 of the video ?

@legendarykamal I checked again it's downloading only once. I have pushed more fixes here, feel free to test and report any bugs.

**Comment by Chadha93** [comment]
@asubedy @legendarykamal Will you test this pr and try to find and report any bugs if facing?  Do perform all the CRUD operations.
Let's make sure we don't have any bugs left in configurations.

**Comment by legendarykamal** [comment]
yes Ok I will be testing it out on my local pc today :+1:

**Pull request merged**